protocol/utils.go

```diff
@@ -36,3 +38,17 @@ func ConvertContractSwapToMsg(swap contracts.SwapFactorySwap) *message.ContractS
 		Nonce:        swap.Nonce,
 	}
 }
+
+// AssetSymbol returns the symbol for the given asset.
+func AssetSymbol(b backend.Backend, asset types.EthAsset) (string, error) {
+	if asset != types.EthAssetETH {
+		_, symbol, _, err := b.ERC20Info(b.Ctx(), asset.Address())
+		if err != nil {
+			return "", fmt.Errorf("failed to get ERC20 info: %w", err)
+		}
+
+		return symbol, nil
+	}
+
+	return "ETH", nil
```

Comment:
I don't feel strongly on this, it's more of a question: Do we need a constant for this?

Reply:
I added one anyways, why not